### PR TITLE
Rate limiting apisix

### PR DESCRIPTION
**Pull request by Arsnael**
Need to test it...

### CODE REVIEW

File: demo/apisix/conf/apisix.yaml [hunk 1]
```diff
@@ -105,4 +112,11 @@ plugin_configs:
         status_code: 200
         body: "APISIX already up!\n"  #For test, TODO remove
     desc: "response-rewrite"
+  -
+    id: limit-req-plugin
+    plugins:
+      limit-req:
+        rate: 100
+        burst: 50
```
[the remark this answers is not in the document]
Comment (Arsnael) [reply demo/apisix/conf/apisix.yaml:120]: Is it so hard to go read the apisix doc? But sure...

**Comment by Arsnael** [comment]
[apisix-rate-limiting.webm](https://user-images.githubusercontent.com/9005025/234543875-fbb4f92a-453b-4f54-bea8-79d9becb14b3.webm)

I did put rate limiter of apisix to 10 request per second and a burst of 5.
I then tinkered quite a bunch gatling simulations to be able to generate some load. I put 6 times the same user in the csv file, then modified the EmailSubmission jmap simulation to generate enough load and use the csv file, also modified the jmap api calls to not use basic auth but oidc token in an Authorization header. I pretty much launched the test.sh script, took the oidc token generated, export it as an env file in bash, import it in the scala jmap api calls.

Then I run it and the result is what can be seen on the video.

File: demo/apisix/conf/apisix.yaml [hunk 2]
```diff
@@ -105,4 +127,11 @@ plugin_configs:
         status_code: 200
         body: "APISIX already up!\n"  #For test, TODO remove
     desc: "response-rewrite"
+  -
+    id: limit-req-plugin
```
Comment (vttranlina) [inline demo/apisix/conf/apisix.yaml:131]: Should we rename it to another, eg: `jmap-plugin`
We will have another config for another plugin. E.g: circuit breaker for this id. 
With the syntax above, not sure `plugin_config_id` will accept several ids inline,

Comment (Arsnael) [reply demo/apisix/conf/apisix.yaml:131]: i agree it's better to regroup plugins common to all routes under one id. I'm ok with your suggestion :)

**Review by vttranlina — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**